metricflow-semantics/metricflow_semantics/dag/mf_dag.py

```diff
@@ -57,7 +57,7 @@ def visit_node(self, node: DagNode) -> VisitorOutputT:  # noqa: D102
 DagNodeT = TypeVar("DagNodeT", bound="DagNode")
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, eq=False)
```

Comment:
Can we add a comment here explaining essentially what you explained in the PR description? Will likely be helpful to future devs.

Reply:
Added.